internal/monitors/haproxy/helpers.go

```diff
@@ -0,0 +1,280 @@
+	"check_duration":     gaugeCheckDuration,
+	"conn_rate":          gaugeConnectionRate,
+	"conn_rate_max":      gaugeConnectionRateMax,
+	"CurrConns":          gaugeCurrentConnections,
+	"CurrSslConns":       gaugeCurrentSslConnections,
+	"dcon":               gaugeDeniedTCPConnections,
+	"dses":               gaugeDeniedTCPSessions,
+	"Idle_pct":           gaugeIdlePct,
+	"intercepted":        gaugeInterceptedRequests,
+	"lastsess":           gaugeLastSession,
+	"MaxConnRate":        gaugeMaxConnectionRate,
+	"MaxConn":            gaugeMaxConnections,
+	"MaxPipes":           gaugeMaxPipes,
+	"MaxSessRate":        gaugeMaxSessionRate,
+	"MaxSslConns":        gaugeMaxSslConnections,
+	"PipesFree":          gaugePipesFree,
+	"PipesUsed":          gaugePipesUsed,
+	"qcur":               gaugeQueueCurrent,
+	"qlimit":             gaugeQueueLimit,
+	"qmax":               gaugeQueueMax,
+	"qtime":              gaugeQueueTimeAvg,
+	"req_rate":           gaugeRequestRate,
+	"req_rate_max":       gaugeRequestRateMax,
+	"rtime":              gaugeResponseTimeAvg,
+	"Run_queue":          gaugeRunQueue,
+	"scur":               gaugeSessionCurrent,
+	"rate":               gaugeSessionRate,
+	"SessRate":           gaugeSessionRateAll,
+	"rate_lim":           gaugeSessionRateLimit,
+	"rate_max":           gaugeSessionRateMax,
+	"ttime":              gaugeSessionTimeAverage,
+	"SslBackendKeyRate":  gaugeSslBackendKeyRate,
+	"SslFrontendKeyRate": gaugeSslFrontendKeyRate,
+	"SslRate":            gaugeSslRate,
+	"Tasks":              gaugeTasks,
+	"throttle":           gaugeThrottle,
+	"ZlibMemUsage":       gaugeZlibMemUsage,
+}
+
+// Creates datapoints by fetching csv stats from an http endpoint.
+func newStatsPageDps(conf *Config, proxies map[string]bool) []*datapoint.Datapoint {
+	body, err := csvReader(conf)
+	if err != nil {
+		logger.Errorf("cannot scrape HAProxy: %v", err)
+		return nil
+	}
+	return newStatsDps(body, proxies)
+}
+
+// Creates datapoints by running the show stats command.
+func newStatsCmdDps(u *url.URL, timeout time.Duration, proxies map[string]bool) []*datapoint.Datapoint {
+	body, err := cmdReader(u, "show stat\n", timeout)
+	if err != nil {
+		logger.Errorf("cannot scrape HAProxy: %v", err)
+	}
+	return newStatsDps(body, proxies)
+}
+
+// Creates datapoints by reading csv stats.
+func newStatsDps(body io.ReadCloser, proxiesToMonitor map[string]bool) []*datapoint.Datapoint {
+	dps := make([]*datapoint.Datapoint, 0)
+	for _, metricValuePairs := range statsMetricValuePairs(body) {
+		if len(proxiesToMonitor) != 0 && !proxiesToMonitor[metricValuePairs["pxname"]] && !proxiesToMonitor[metricValuePairs["svname"]] {
+			continue
+		}
+		for metric, value := range metricValuePairs {
+			if dp := newDp(sfxMetricsMap[metric], value); dp != nil {
+				dp.Dimensions["proxy_name"] = metricValuePairs["pxname"]
+				dp.Dimensions["service_name"] = metricValuePairs["svname"]
+				dp.Dimensions["process_num"] = pidPlusPlus(metricValuePairs["pid"])
```

Comment:
Can you put a comment explaining this logic of incrementing the PID.  It isn't clear at all.

Reply:
ok